### PR TITLE
Update README file for 0.9.0 release

### PR DESCRIPTION
**Pull request by zhangguanheng66**

### CODE REVIEW

File: torchtext/legacy/README.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+In v0.9.0 release, we move the following legacy code to `torchtext.legacy <#legacy>`_. This is part of the work to revamp the torchtext library and the motivation has been discussed in `Issue #664 <https://github.com/pytorch/text/issues/664>`_:
+
+* ``torchtext.legacy.data.field``
+* ``torchtext.legacy.data.batch``
+* ``torchtext.legacy.data.example``
+* ``torchtext.legacy.data.iterator``
+* ``torchtext.legacy.data.pipeline``
+* ``torchtext.legacy.datasets``
+
+We have a `migration tutorial <https://fburl.com/9hbq843y>`_ to help users switch to the torchtext datasets in ``v0.9.0`` release. For the users who still want the legacy components, they can add ``legacy`` to the import path.
```
Comment (cpuhrsch) [inline torchtext/legacy/README.rst:13]: fburls are not public

Comment (cpuhrsch) [reply torchtext/legacy/README.rst:13]: You can export a colab notebook as an ipython notebook and upload the ipython notebook here.

Then you can create a colab link that uses the ipython on github like so:
```
https://colab.research.google.com/github/pytorch/nestedtensor/blob/master/tutorials/notebooks/basic.ipynb
```

Comment (zhangguanheng66) [reply torchtext/legacy/README.rst]: Move the tutorial to `examples/migration_tutorial` folder https://colab.research.google.com/github/zhangguanheng66/text/blob/migration_tutorial/examples/legacy_tutorial/migration_tutorial.ipynb

File: torchtext/legacy/README.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+In v0.9.0 release, we move the following legacy code to `torchtext.legacy <#legacy>`_. This is part of the work to revamp the torchtext library and the motivation has been discussed in `Issue #664 <https://github.com/pytorch/text/issues/664>`_:
+
+* ``torchtext.legacy.data.field``
+* ``torchtext.legacy.data.batch``
+* ``torchtext.legacy.data.example``
+* ``torchtext.legacy.data.iterator``
+* ``torchtext.legacy.data.pipeline``
+* ``torchtext.legacy.datasets``
+
+We have a `migration tutorial <https://colab.research.google.com/github/zhangguanheng66/text/blob/master/examples/legacy_tutorial/migration_tutorial.ipynb>`_ to help users switch to the torchtext datasets in ``v0.9.0`` release. For the users who still want the legacy components, they can add ``legacy`` to the import path.
```
Comment (cpuhrsch) [inline torchtext/legacy/README.rst:13]: You probably meant to link to pytorch/text instead of your fork

File: README.rst [hunk 3]
```diff
@@ -12,14 +12,12 @@ torchtext
 
 This repository consists of:
 
-* `torchtext.data <#data>`_: Generic data loaders, abstractions, and iterators for text (including vocabulary and word vectors)
-* `torchtext.datasets <#datasets>`_: Pre-built loaders for common NLP datasets
+* `torchtext.datasets <#datasets>`_: The raw text iterators for common NLP datasets
+* `torchtext.data <#data>`_: Some basic NLP building blocks (tokenizers, metrics, functionals etc.)
+* `torchtext.nn <#nn>`_: NLP related modules
```
Comment (parmeet) [inline README.rst:17]: here also about the link

File: README.rst [hunk 4]
```diff
@@ -12,14 +12,12 @@ torchtext
...
-
-    pip install --pre torch torchtext -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html  
-
-For more detailed instructions, please refer to `Install PyTorch <https://pytorch.org/get-started/locally/>`_. It should be noted that the new building blocks are still under development, and the APIs have not been solidified.
+Note: the legacy code discussed in `torchtext v0.7.0 release note <https://github.com/pytorch/text/releases/tag/v0.7.0-rc3>`_ has been retired to `torchtext.legacy <#legacy>`_ folder. Those legacy code will not be maintained by the development team, and we plan to fully remove them in the future release. See the Legacy session for more details.
```
Comment (parmeet) [inline README.rst:20]: same comment about the link for legacy folder

File: README.rst [hunk 5]
```diff
@@ -12,14 +12,12 @@ torchtext
...
 
 Installation
```
Comment (parmeet) [inline README.rst:22]: In line below, I think Anaconda not the package manager but is more like a distribution whereas conda is the underlying package manager. Should we rephrase it to something like: "We recommend conda as package manager that comes with Anaconda or miniconda"?

File: README.rst [hunk 6]
```diff
@@ -12,14 +12,12 @@ torchtext
 
 This repository consists of:
 
-* `torchtext.data <#data>`_: Generic data loaders, abstractions, and iterators for text (including vocabulary and word vectors)
-* `torchtext.datasets <#datasets>`_: Pre-built loaders for common NLP datasets
+* `torchtext.datasets <#datasets>`_: The raw text iterators for common NLP datasets
```
[the remark this answers is not in the document]
Comment (zhangguanheng66) [reply README.rst]: Good catch. Fixed them all.

File: README.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhangguanheng66) [reply README.rst]: It's actually "folder"

File: README.rst (hunk 5 above)
Comment (zhangguanheng66) [reply README.rst:22]: It's very general statement. I think we can "We recommend Anaconda to access packages"

File: README.rst [hunk 7]
```diff
@@ -88,105 +87,62 @@ Documentation
...
-* Machine translation: abstract class + Multi30k, IWSLT, WMT14
-* Sequence tagging (e.g. POS/NER): abstract class + UDPOS, CoNLL2000Chunking
-* Question answering: 20 QA bAbI tasks
-* Text classification: AG_NEWS, SogouNews, DBpedia, YelpReviewPolarity, YelpReviewFull, YahooAnswers, AmazonReviewPolarity, AmazonReviewFull
+We have re-written several building blocks under ``torchtext.experimental``:
 
-Others are planned or a work in progress:
+* Transforms
+* Vocabulary
+* Vectors
```
Comment (parmeet) [inline README.rst:126]: Should we provide short note or one liner description of each of "Transforms", "Vocabulary" and "Vectors". Or at-least a link where users can learn more about it?  Since these are prototype features where we probably want some feedback, users might want to learn about them quickly through somewhere?

Comment (zhangguanheng66) [reply README.rst:126]: Sure. Let me add those links and some descriptions.

File: README.rst (hunk 5 above)
Comment (parmeet) [reply README.rst:22]: Sure, we can probably leave it as such then to keep the consistency with PyTorch as this is a minor technical detail that users won't mind :)

@cpuhrsch  Just FYI

**Review by parmeet — APPROVED** [review]
